core/src/state.rs

```diff
@@ -899,7 +865,34 @@ impl WorldTransaction<'_, '_> {
 
     /// Set executor data model.
     pub fn set_executor_data_model(&mut self, executor_data_model: ExecutorDataModel) {
-        *self.executor_data_model.get_mut() = executor_data_model;
+        let prev_executor_data_model =
+            core::mem::replace(self.executor_data_model.get_mut(), executor_data_model);
+
+        self.update_parameters(&prev_executor_data_model);
+    }
+
+    fn update_parameters(&mut self, prev_executor_data_model: &ExecutorDataModel) {
+        let removed_parameters = prev_executor_data_model
+            .parameters
+            .keys()
+            .filter(|param_id| !self.executor_data_model.parameters.contains_key(param_id));
+        let new_parameters = self
+            .executor_data_model
+            .parameters
+            .iter()
+            .filter(|(param_id, _)| !prev_executor_data_model.parameters.contains_key(param_id));
+
+        for param in removed_parameters {
+            iroha_logger::info!("{}: parameter removed", param);
```

Comment:
maybe `debug` or even `trace` level? and below as well

Reply:
this will be quite a rare log entry, so I think `INFO` is adequate